tests/unifiedPlatform/pds2/pds_dataservice_test.go

```diff
@@ -355,3 +359,99 @@ var _ = Describe("{ValidatePdsHealthIncaseofFailures}", func() {
 		defer EndPDSTorpedoTest()
 	})
 })
+
+var _ = Describe("{ScaleDownScaleupPXCluster}", func() {
+	var (
+		deployment *automationModels.PDSDeploymentResponse
+		err        error
+		nodeList   []*corev1.Node
+	)
+
+	JustBeforeEach(func() {
+		StartPDSTorpedoTest("ScaleDownScaleupPXCluster", "Scales Down the PX cluster, Verify Data Services and Scales up the Px cluster", nil, 0)
+	})
+
+	It("Scales Down the PX cluster, Verify Data Services and Scales up the Px cluster", func() {
+
+		// Getting all worker nodes
+		workerNodes := node.GetWorkerNodes()
+
+		for _, ds := range NewPdsParams.DataServiceToTest {
+			Step("Deploy DataService", func() {
+				deployment, err = WorkflowDataService.DeployDataService(ds, ds.Image, ds.Version, PDS_DEFAULT_NAMESPACE)
+				log.FailOnError(err, "Error while deploying ds")
+				log.Debugf("Source Deployment Id: [%s]", *deployment.Create.Meta.Uid)
+				nodes, err := pdslib.GetNodesOfSS(*deployment.Create.Status.CustomResourceName, PDS_DEFAULT_NAMESPACE)
+				log.FailOnError(err, "Error while getting Data Serice Nodes")
+				nodeList = append(nodeList, nodes[0])
+
+			})
+
+			Step("Scale down PX Nodes on which Data Services are running", func() {
+				// disable PX Pod on the first node of each deployed Data Service
+				for _, nodeToStop := range nodeList {
```

Comment:
can we write a function to stop px on a node ?

Reply:
Done